packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -519,9 +524,11 @@ export class CfnParser {
     if (isRef) {
       const refElement = this.options.finder.findRefTarget(refTarget);
       if (!refElement) {
-        throw new Error(`Element referenced in Fn::Sub expression with logical ID: '${refTarget}' was not found in the template`);
+        if  (!(refTarget in this.parameters)) {
+          throw new Error(`Element referenced in Fn::Sub expression with logical ID: '${refTarget}' was not found in the template`);
+        }
       }
-      return leftHalf + CfnReference.for(refElement, 'Ref', true).toString() + this.parseFnSubString(rightHalf, map);
+      return leftHalf + CfnReference.for(refElement!, 'Ref', true).toString() + this.parseFnSubString(rightHalf, map);
```

Comment:
If you undo the weird change from above, this `!` will not be required anymore, and should also be reverted.